src/sentry/uptime/consumers/results_consumer.py

```diff
@@ -17,19 +18,38 @@
     ResultsStrategyFactory,
 )
 from sentry.uptime.detectors.ranking import _get_cluster
+from sentry.uptime.detectors.tasks import set_failed_url
 from sentry.uptime.issue_platform import create_issue_platform_occurrence
-from sentry.uptime.models import ProjectUptimeSubscription, UptimeSubscription
+from sentry.uptime.models import (
+    ProjectUptimeSubscription,
+    ProjectUptimeSubscriptionMode,
+    UptimeSubscription,
+)
+from sentry.uptime.subscriptions.subscriptions import (
+    create_uptime_subscription,
+    delete_project_uptime_subscription,
+    remove_uptime_subscription_if_unused,
+)
 from sentry.utils import metrics
 from sentry.utils.hashlib import md5_text
 
 logger = logging.getLogger(__name__)
 LAST_UPDATE_REDIS_TTL = timedelta(days=7)
+ONBOARDING_FAILURE_THRESHOLD = 3
+ONBOARDING_FAILURE_REDIS_TTL = timedelta(days=2)
+ONBOARDING_MONITOR_PERIOD = timedelta(days=3)
+# How frequently we should run active auto-detected subscriptions
+AUTO_DETECTED_ACTIVE_SUBSCRIPTION_INTERVAL_SECONDS = int(timedelta(minutes=5).total_seconds())
 
 
 def build_last_update_key(project_subscription: ProjectUptimeSubscription) -> str:
     return f"project-sub-last-update:{md5_text(project_subscription.id).hexdigest()}"
 
 
+def build_onboarding_failure_key(project_subscription: ProjectUptimeSubscription) -> str:
+    return f"project-sub-onboarding_failure:{md5_text(project_subscription.id).hexdigest()}"
```

Comment:
Why med5 this? Can't we just use the hex value of the uuid?
```suggestion
    return f"project-sub-onboarding_failure:{project_subscription.id.hex}"
```

Reply:
I'm just generally in the habit of md5ing things, but this will work too

Reply:
Wait no - this is just an int, not the uuid

Reply:
oh, can we just use the int then?